onchain/src/tests/hodl_quest.cairo

```diff
@@ -0,0 +1,67 @@
+use art_peace::{IArtPeaceDispatcher, IArtPeaceDispatcherTrait};
+use art_peace::quests::hodl_quest::HodlQuest::HodlQuestInitParams;
+use art_peace::quests::interfaces::{IQuestDispatcher, IQuestDispatcherTrait};
+
+use art_peace::tests::art_peace::deploy_with_quests_contract;
+use art_peace::tests::utils;
+use starknet::{ContractAddress, get_caller_address, get_contract_address, contract_address_const};
+use snforge_std as snf;
+use snforge_std::{declare, CheatTarget, ContractClassTrait};
+
+const reward_amt: u32 = 18;
+const user_extra_pixel: u32 = 0;
+
+
+fn deploy_hodl_quest() -> ContractAddress {
+    let contract = declare("HodlQuest");
+
+    let mut hodl_quest_calldata = array![];
+    HodlQuestInitParams {
+        art_peace: utils::ART_PEACE_CONTRACT(), reward: reward_amt, extra_pixel: user_extra_pixel,
+    }
+        .serialize(ref hodl_quest_calldata);
+
+    contract.deploy(@hodl_quest_calldata).unwrap()
+}
+
+
+#[test]
+fn deploy_hodl_quest_test() {
+    let hodl_quest = deploy_hodl_quest();
+    let art_peace = IArtPeaceDispatcher {
+        contract_address: deploy_with_quests_contract(array![].span(), array![hodl_quest].span())
+    };
+
+    let zero_address = contract_address_const::<0>();
+
+    assert!(
+        art_peace.get_days_quests(0) == array![zero_address, zero_address, zero_address].span(),
+        "Daily quests were not set correctly"
+    );
+    assert!(
+        art_peace.get_main_quests() == array![hodl_quest].span(),
+        "Main quests were not set correctly"
+    );
+}
+
+
+#[test]
+fn hodl_quest_test() {
+    let hodl_quest_contract_address = deploy_hodl_quest();
+
+    let art_peace = IArtPeaceDispatcher {
+        contract_address: deploy_with_quests_contract(
+            array![].span(), array![hodl_quest_contract_address].span()
+        )
+    };
+
+    let calldata: Array<felt252> = array![];
+
```

Comment:
Please add `start_prank` to specify the user address, and use `store` to update the users extra pixel count to `user_extra_pixel`.

Refs
- [Store docs](https://foundry-rs.github.io/starknet-foundry/appendix/cheatcodes/store.html?highlight=store#store)
- [Start Prank](https://github.com/keep-starknet-strange/art-peace/blob/294e956d821978aa9e4a62a4ebd51c77134f8dd5/onchain/src/tests/nft_quest.cairo#L66)
- [Store example](https://github.com/keep-starknet-strange/blobstream-starknet/blob/842ee5ad46dede70ec8bf5d1615426e44e3e08cc/src/verifier/tests/test_verifier.cairo#L62)

Reply:
@b-j-roberts kindly review the changes